modules/collection/programs/direnv.nix

```
@@ -0,0 +1,106 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}: let
+  inherit (lib.meta) getExe;
+  inherit (lib.modules) mkIf;
+  inherit (lib.options) mkEnableOption mkOption mkPackageOption;
+  inherit (lib.types) bool lines;
+
+  toml = pkgs.formats.toml {};
+
+  cfg = config.rum.programs.direnv;
+in {
+  options.rum.programs.direnv = {
+    enable = mkEnableOption "direnv";
+
+    package = mkPackageOption pkgs "direnv" {};
+
+    settings = mkOption {
+      type = toml.type;
+      default = {};
+      example = {
+        whitelist.prefix = ["~/src"];
+      };
+      description = ''
+        Configuration written to {file}`$HOME/.config/direnv/direnv.toml`.
+        Please reference [direnv's documentation] for config options.
+
+        [direnv's documentation]: https://direnv.net/man/direnv.toml.1.html
+      '';
+    };
+
+    direnvrc = mkOption {
+      type = lines;
+      default = "";
+      example = ''
+        : ''${XDG_CACHE_HOME:=$HOME/.cache}
+        declare -A direnv_layout_dirs
+        direnv_layout_dir() {
+        	echo "''${direnv_layout_dirs[$PWD]:=$(
+        		echo -n "$XDG_CACHE_HOME"/direnv/layouts/
+        		echo -n "$PWD" | sha1sum | cut -d ' ' -f 1
+        	)}"
+        }
+      '';
+      description = ''
+        Bash code loaded before every `.envrc`. Good for personal extensions.
+        Find community-maintained examples in [direnv's wiki]
+
+        [direnv's wiki]: https://github.com/direnv/direnv/wiki
+      '';
+    };
+
+    integrations = {
+      fish.enable = mkOption {
+        type = bool;
+        default = false;
+        example = true;
+        description = ''
+          Whether to enable direnv integration with fish.
+        '';
+      };
+
+      nix-direnv.enable = mkOption {
+        type = bool;
+        default = false;
+        example = true;
+        description = ''
+          Whether to enable [nix-direnv] integration with direnv.
+
+          [nix-direnv]: https://github.com/nix-community/nix-direnv
+        '';
+      };
+      nix-direnv.package = mkPackageOption pkgs "nix-direnv" {};
```

I'm gonna say no on the auto-enabling the nix-direnv integration, as we don't want to make the user think something is a native component of direnv rather than nix-direnv, and it should be pretty clear for users to see the nix-direnv option as well.

```suggestion
    integrations = {
      fish.enable = mkEnableOption "direnv integration with fish";
      nix-direnv = {
        enable = mkEnableOption "direnv integration with nix-direnv";
        package = mkPackageOption pkgs "nix-direnv" {};
      };
```

(Just a note: if you want to automatically set an enable option to default with mkEnableOption, here's a tricky little way to do that:)

```nix
myOption = mkEnableOption "a special program (enabled by default)" // {
  default = true;
};
```